ucan/src/capability/semantics.rs

```diff
@@ -1,137 +1,85 @@
-        CapabilityIpld {
-            with: capability.with().to_string(),
-            can: capability.can().to_string(),
-
-            // TODO(#22): Full support for 0.9 and the nb field
-            nb: None,
-        }
-    }
-}
-
-impl TryFrom<&Value> for CapabilityIpld {
-    type Error = anyhow::Error;
-
-    fn try_from(value: &Value) -> Result<Self, Self::Error> {
-        match value {
-            Value::Object(map) => {
-                let with = map
-                    .get("with")
-                    .ok_or_else(|| anyhow!("Missing 'with' field"))?;
-                let can = map
-                    .get("can")
-                    .ok_or_else(|| anyhow!("Missing 'can' field"))?;
-                let nb = map.get("nb").cloned();
-
-                let with = match with {
-                    Value::String(with) => with.clone(),
-                    _ => return Err(anyhow!("The 'with' field must be a string")),
-                };
-
-                let can = match can {
-                    Value::String(can) => can.to_lowercase(),
-                    _ => return Err(anyhow!("The 'can' field must be a string")),
-                };
-
-                Ok(CapabilityIpld { with, can, nb })
-            }
-            _ => Err(anyhow!("Not a valid capability: {}", value)),
-        }
-    }
-}
-
 pub trait Scope: ToString + TryFrom<Url> + PartialEq + Clone {
     fn contains(&self, other: &Self) -> bool;
 }
 
 pub trait Action: Ord + TryFrom<String> + ToString + Clone {}
 
 #[derive(Clone, Eq, PartialEq)]
-pub enum Resource<S>
+pub enum ResourceUri<S>
 where
     S: Scope,
 {
     Scoped(S),
     Unscoped,
 }
 
-impl<S> Resource<S>
+impl<S> ResourceUri<S>
 where
     S: Scope,
 {
     pub fn contains(&self, other: &Self) -> bool {
         match self {
-            Resource::Unscoped => true,
-            Resource::Scoped(scope) => match other {
-                Resource::Scoped(other_scope) => scope.contains(other_scope),
+            ResourceUri::Unscoped => true,
+            ResourceUri::Scoped(scope) => match other {
+                ResourceUri::Scoped(other_scope) => scope.contains(other_scope),
                 _ => false,
             },
         }
     }
 }
 
-impl<S> ToString for Resource<S>
+impl<S> ToString for ResourceUri<S>
 where
     S: Scope,
 {
     fn to_string(&self) -> String {
         match self {
-            Resource::Unscoped => "*".into(),
-            Resource::Scoped(value) => value.to_string(),
+            ResourceUri::Unscoped => "*".into(),
+            ResourceUri::Scoped(value) => value.to_string(),
         }
     }
 }
 
 #[derive(Clone, Eq, PartialEq)]
-pub enum With<S>
+pub enum Resource<S>
 where
     S: Scope,
 {
-    Resource { kind: Resource<S> },
-    My { kind: Resource<S> },
-    As { did: String, kind: Resource<S> },
+    Resource { kind: ResourceUri<S> },
+    My { kind: ResourceUri<S> },
+    As { did: String, kind: ResourceUri<S> },
```

Comment:
I think `my:` and `as:` have been removed from the spec. Meanwhile, there is a new [`ucan:`](https://github.com/ucan-wg/spec#41-ucan). What are your thoughts on how to approach accommodating this change?

Reply:
Can you file an issue to track this additional work?

Reply:
I've been using https://github.com/ucan-wg/rs-ucan/issues/25 to track, can we update the issue name?